vitaliy_dobrovichan/task2/extractor_tool.rb

```diff
@@ -0,0 +1,81 @@
+class ExtractorTool
+  attr_reader   :name, :browser
+  attr_accessor :result
+
+  def initialize(browser, name)
+    @browser = browser
+    @name    = name
+    @result  = {}
+  end
+
+  def result_page_link(class_name)
+    @browser.link(class: class_name)
+  end
+
+  def english_name
+    @browser.h1(class: "moviename-big").following_sibling.text
+  end
+
+  def table
+    @browser.table(class: "info")
+  end
+
+  def year
+    table.td(text: "год").following_sibling.text.to_i
+  end
+
+  def producers
+    table.td(text: "продюсер").following_sibling.text.split(",")[0...-1]
+  end
+
+  def jenres
+    table.td(text: "жанр").following_sibling.text.split(",")[0...-1]
+  end
+
+  def actors
+    @browser.div(id: "actorList").ul.lis.map {|li| li.text }[0...-1]
+  end
+
+  def poster_source
+    @browser.a(class: "popupBigImage").img.src
+  end
+
+  def assign_attributes
+    @result[:name]          = english_name
+    @result[:year]          = year
+    @result[:producers]     = producers
+    @result[:jenres]        = jenres
+    @result[:actors]        = actors
+    @result[:poster_source] = poster_source
+
+    OpenStruct.new(
+      name:          result[:name],
+      year:          result[:year],
+      producers:     result[:producers],
+      jenres:        result[:jenres],
+      actors:        result[:actors],
+      poster_source: result[:poster_source]
+    )
+  end
+
+  def build_result
+    link = result_page_link("header-search-suggest-partial-component__item_type_first")
+
+    begin
+      link.wait_until_present(timeout: 3)
+    rescue
```

Comment:
do not rescue without implicit exception, it will catch any exception inherited from `Exception` class (every exception is), therefore you'll not able even to interrupt your code as it will be caught by rescue ;)
use `rescue StandardError` at least, but better catch only errors you expect
[check](https://stackoverflow.com/questions/10048173/why-is-it-bad-style-to-rescue-exception-e-in-ruby)